### PR TITLE
fix: re-render of address list

### PR DESCRIPTION
**Pull request by Sinucid**
Checked attribute sets only the default state and doesn't affect on property, but LIT is able to manages it under the hood and boolean checked attribute changes the prop. But for some cases, like deletion of currently checked radio button, this alignments doesn't work. To overcome this issue `keyed` directive is using for address list component, to trigger re-render and align attribute and property correctly 

closes to [HRZ-2237](https://spryker.atlassian.net/browse/HRZ-2237)

### CODE REVIEW

File: libs/domain/user/address-list/src/address-list.component.ts [hunk 1]
```diff
@@ -46,9 +49,17 @@ export class AddressListComponent extends ComponentMixin<AddressListItemOptions>
     distinctUntilChanged()
   );
 
-  protected addresses$ = combineLatest([
-    resolve(AddressService).getAddresses().pipe(distinctUntilChanged()),
-    new ContentController(this).getOptions(),
+  protected addresses$ = this.addressService
+    .getAddresses()
+    .pipe(distinctUntilChanged());
```
Comment (supproduction) [inline libs/domain/user/address-list/src/address-list.component.ts]: `distinctUntilChanged` doesn't make sense for object

File: libs/domain/user/address-list/src/address-list.component.ts [hunk 2]
```diff
@@ -46,9 +49,17 @@ export class AddressListComponent extends ComponentMixin<AddressListItemOptions>
     distinctUntilChanged()
```
Comment (supproduction) [inline libs/domain/user/address-list/src/address-list.component.ts]: Why do we need `distinctUntilChanged` here?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/user/address-list/src/address-list.component.ts (hunk 1 above)
Comment (Sinucid) [reply libs/domain/user/address-list/src/address-list.component.ts]: Yeah, my bad. I refactored it but didn't pay attention on observables logic

File: libs/domain/user/address-list/src/address-list.component.ts (hunk 2 above)
Comment (Sinucid) [reply libs/domain/user/address-list/src/address-list.component.ts]: Not necessary, dropped

**Review by supproduction — APPROVED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**